apps/system/js/statusbar.js

```diff
@@ -14,597 +14,567 @@
-        if (icon.name === 'OperatorIcon') {
-          this.updateOperatorWidth(icon);
-        }
-        break;
-
-      case 'lockscreen-appopened':
-        // XXX: Use Service.query('getTopMostWindow')
-        this._inLockScreenMode = true;
-        this.setAppearance();
-        break;
-
-      case 'lockscreen-appclosing':
-        this._inLockScreenMode = false;
-        this.setAppearance();
-        break;
-
-      case 'attentionopened':
-        this.element.classList.add('maximized');
-        this.element.classList.remove('light');
-        break;
-
-      case 'sheets-gesture-begin':
-        this.element.classList.add('hidden');
-        if (!this._pausedForGesture) {
+      } else {
+        window.addEventListener('ftuskip', this);
+        window.addEventListener('ftudone', this);
+      }
+      Service.registerState('height', this);
+      Service.register('pauseUpdate', this);
+    },
+
+    /**
+     * Finish all initializing statusbar event handlers
+     */
+    finishInit: function() {
+      window.addEventListener('sheets-gesture-begin', this);
+      window.addEventListener('sheets-gesture-end', this);
+      window.addEventListener('utilitytraywillshow', this);
+      window.addEventListener('utilitytraywillhide', this);
+      window.addEventListener('utility-tray-overlayopened', this);
+      window.addEventListener('utility-tray-overlayclosed', this);
+      window.addEventListener('utility-tray-abortopen', this);
+      window.addEventListener('utility-tray-abortclose', this);
+      window.addEventListener('cardviewshown', this);
+      window.addEventListener('cardviewclosed', this);
+
+      // Listen to 'lockscreen-appopened', 'lockscreen-appclosing', and
+      // 'lockpanelchange' in order to correctly set the visibility of
+      // the statusbar clock depending on the active lockscreen panel
+      window.addEventListener('lockscreen-appopened', this);
+      window.addEventListener('lockscreen-appclosing', this);
+      window.addEventListener('lockpanelchange', this);
+
+      // Listen to orientation change and SHB activation/deactivation.
+      window.addEventListener('system-resize', this);
+
+      window.addEventListener('attentionopened', this);
+      window.addEventListener('appopening', this);
+      window.addEventListener('appopened', this);
+      window.addEventListener('hierarchytopmostwindowchanged', this);
+      window.addEventListener('activityopened', this);
+      window.addEventListener('activitydestroyed', this);
+      window.addEventListener('homescreenopening', this);
+      window.addEventListener('homescreenopened', this);
+      window.addEventListener('stackchanged', this);
+
+      // Listen to updates dialog
+      window.addEventListener('updatepromptshown', this);
+      window.addEventListener('updateprompthidden', this);
+
+      // We need to preventDefault on mouse events until
+      // https://bugzilla.mozilla.org/show_bug.cgi?id=1005815 lands
```

Comment:
Can we remove this now? https://bugzilla.mozilla.org/show_bug.cgi?id=1005815#c38